src/applications/gqrx/receiver.cpp

```diff
@@ -180,14 +180,29 @@ void receiver::stop()
  */
 void receiver::set_input_device(const std::string device)
 {
-    qDebug() << "Set input device:";
-    qDebug() << "  old:" << input_devstr.c_str();
-    qDebug() << "  new:" << device.c_str();
-
     std::string error = "";
 
     if (device.empty())
         return;
+    if (input_devstr.compare(device) == 0)
+    {
+#ifndef QT_NO_DEBUG_OUTPUT
```

Comment:
Why is `#ifndef QT_NO_DEBUG_OUTPUT` needed? Doesn't setting that disable `qDebug()` automatically?

Reply:
I don't know where did I took this from. No useful information in case, the input device did not change. So, reverted to previous state and added a check for changed device string as it was done before, but removed later for some unknown purpose.
Why should we always reopen a device even if it did not change? If we have to work around some buggy device, then it is better to check for it's specific device string and reopen if it matches.
Moved this to https://github.com/gqrx-sdr/gqrx/pull/1021.